app/src/main/java/org/geekhaven/ishare/activities/LoginActivity.kt

```diff
@@ -0,0 +1,100 @@
+package org.geekhaven.ishare.activities
+
+import android.content.Intent
+import android.os.Bundle
+import android.support.v7.app.AppCompatActivity
+import android.support.v7.widget.Toolbar
+import android.text.TextUtils
+import android.view.View
+import android.widget.Button
+import android.widget.EditText
+import android.widget.ProgressBar
+import android.widget.Toast
+
+import com.google.android.gms.tasks.OnCompleteListener
+import com.google.android.gms.tasks.Task
+import com.google.firebase.auth.AuthResult
+import com.google.firebase.auth.FirebaseAuth
+
+import org.geekhaven.ishare.R
+
+class LoginActivity : AppCompatActivity() {
+
+    private var inputEmail: EditText? = null
+    private var inputPassword: EditText? = null
+    private var auth: FirebaseAuth? = null
+    private var progressBar: ProgressBar? = null
+    private var btnSignup: Button? = null
+    private var btnLogin: Button? = null
+    private var btnReset: Button? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        //Get Firebase auth instance
+        auth = FirebaseAuth.getInstance()
+
+        if (auth!!.currentUser != null) {
+            startActivity(Intent(this@LoginActivity, MainActivity::class.java))
+            finish()
+        }
+
+        // set the view now
+        setContentView(R.layout.activity_signin)
+
+//        val toolbar = findViewById<View>(R.id.toolbar) as Toolbar
+//        setSupportActionBar(toolbar)
+
+        inputEmail = findViewById<View>(R.id.email) as EditText
```

Comment:
is there some specific reason why you assigned variables for edittexts and buttons? (In kotlin you can directly access them by their IDs.)